src/pool/mempool.rs

```diff
@@ -126,7 +126,7 @@ impl<SP: starknet::providers::Provider + Send + Sync + Clone + 'static> AccountM
         // Convert the optional Ethereum block ID to a Starknet block ID.
         let starknet_block_id = self.eth_client.eth_provider().to_starknet_block_id(Some(BlockId::default())).await?;
         // Create a new Starknet provider wrapper.
-        let starknet_provider = StarknetProvider::new(Arc::new(self.eth_client.eth_provider().starknet_provider()));
+        let starknet_provider = self.eth_client.eth_provider().starknet_provider();
```

Comment:
I feel like we do this a lot, should we have a `starknet_provider` method on the client?

Reply:
Yeah good idea, just fixed